### PR TITLE
Add explicit message ack/nack

### PR DESCRIPTION
**Pull request by necrophonic**

### CODE REVIEW

File: app/consumer.py [hunk 1]
```diff
@@ -13,21 +25,34 @@ def __init__(self):
...
+        except BadMessageError as e:
+            # If it's a bad message then we have to reject it
+            self.reject_message(basic_deliver.delivery_tag, tx_id=processor.tx_id)
+            logger.error("Bad message", action="rejected", exception=e, delivery_count=delivery_count, tx_id=processor.tx_id)
 
-        except Exception as e:
-            logger.error("ResponseProcessor failed", exception=e, tx_id=document['tx_id'])
+        except (RetryableError, Exception) as e:
+            self.nack_message(basic_deliver.delivery_tag, tx_id=processor.tx_id)
+            logger.error("Failed to process", action="nack for retry", exception=e, delivery_count=delivery_count, tx_id=processor.tx_id)
```
Comment (eldeal) [inline app/consumer.py]: Action is 'nack' in some repos and 'nack for retry' in others, if they're not functionally different a consistent message would be preferable (though could be raised as a tech debt issue)

File: app/processors/cora_processor.py [hunk 2]
```diff
@@ -8,10 +9,21 @@ def __init__(self, logger, survey, ftpconn):
...
         self.ftp = ftpconn
 
-    def setup_logger(self):
+    def process(self):
+        transformed = self._transform()
+
+        delivered = self.ftp.unzip_and_deliver(self._get_ftp_folder(self.survey), transformed)
+
+        if not delivered:
+            self.logger.error("Failed to deliver zip to ftp")
```
Comment (eldeal) [inline app/processors/cora_processor.py:21]: The tx_id would be useful here if possible

File: app/processors/cora_processor.py [hunk 3]
```diff
@@ -21,29 +33,34 @@ def setup_logger(self):
...
-        return response.content
+        response = remote_call(endpoint, json=self.survey)
 
-    def deliver_zip(self, zip_contents):
-        folder = self.get_ftp_folder(self.survey)
-        return self.ftp.unzip_and_deliver(folder, zip_contents)
+        if response.status_code == 400 or response.content is None:
+            # There was an issue with the data. This would imply the data was
+            # corrupt or wrong in some way that means it cannot be transformed
+            self.logger.error("Failed to transform", request_url=endpoint, status_code=response.status_code)
```
Comment (eldeal) [inline app/processors/cora_processor.py:52]: tx_id if possible

File: app/processors/cora_processor.py (hunk 2 above)
Comment (necrophonic) [reply app/processors/cora_processor.py:21]: `tx_id` gets bound to the logger during `__init__` via `_setup_logger()`

**Review by eldeal — APPROVED** [review]

**Pull request merged**